exporter/uptraceexporter/translator.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2021 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package uptraceexporter
+
+import (
+	"encoding/binary"
+	"encoding/json"
+
+	"github.com/uptrace/uptrace-go/spanexp"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/otel/label"
+)
+
+func asUint64(b [8]byte) uint64 {
+	return binary.LittleEndian.Uint64(b[:])
+}
+
+func spanKind(kind pdata.SpanKind) string {
+	switch kind {
+	case pdata.SpanKindCLIENT:
+		return "client"
+	case pdata.SpanKindSERVER:
+		return "server"
+	case pdata.SpanKindPRODUCER:
+		return "producer"
+	case pdata.SpanKindCONSUMER:
+		return "consumer"
+	}
+	return "internal"
+}
+
+func statusCode(status pdata.StatusCode) string {
+	switch status {
+	case pdata.StatusCodeOk:
+		return "ok"
+	case pdata.StatusCodeError:
+		return "error"
+	}
+	return "unset"
+}
+
+func keyValueSlice(attrs pdata.AttributeMap) spanexp.KeyValueSlice {
+	out := make(spanexp.KeyValueSlice, 0, attrs.Len())
+
+	attrs.ForEach(func(key string, value pdata.AttributeValue) {
```

Comment:
Was it intentional that pdata.AttributeValueNULL was left out?

Reply:
Yes, Uptrace does not directly support `NULL` values. I could encode `NULL` values as empty string but I am not sure what it solves...

I've changed the code to explicitly ignore NULL values.